arch/common/shared_irq.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sw_isr_table.h>
+
+void shared_isr(const void *data)
+{
+	size_t i;
+	const struct shared_irq_data *irq_data = data;
```

Comment:
any reason to not directly have
```c
void shared_isr(const struct shared_irq_data *irq_data)
```

Reply:
Would probably require a cast each time `_sw_isr_table`'s `isr` field was assigned the value of `shared_isr`. This is not worth the trouble IMO.

Reply:
oh right, nvm.